langchain/vectorstores/marqo.py

```diff
@@ -0,0 +1,413 @@
+        """
+        results = self._client.index(self._index_name).search(q=query, limit=k)
+        return results["hits"]
+
+    @classmethod
+    def from_documents(
+        cls,
+        documents: List[Document],
+        embedding: Any = None,
+        **kwargs: Any,
+    ) -> VectorStore:
+        """Return VectorStore initialized from documents. Note that Marqo does not need embeddings, we retain the
+        parameter to adhere to the Liskov substitution principle.
+
+
+        Args:
+            documents (List[Document]): Input documents
+            embedding (Any, optional): Embeddings (not required). Defaults to None.
+
+        Returns:
+            VectorStore: A Marqo vectorstore
+        """
+        texts = [d.page_content for d in documents]
+        metadatas = [d.metadata for d in documents]
+        return cls.from_texts(texts, metadatas=metadatas, **kwargs)
+
+    @classmethod
+    def from_texts(
+        cls: Marqo,
+        texts: List[str],
+        embedding: Any = None,
+        metadatas: Optional[List[dict]] = None,
+        index_name: str = None,
+        url: str = "http://localhost:8882",
+        api_key: str = "",
+        marqo_device: str = "cpu",
+        add_documents_settings: Optional[Dict[str, Any]] = {},
+        searchable_attributes: Optional[List[str]] = None,
+        index_settings: Optional[Dict[str, Any]] = {},
+        verbose: bool = True,
+        **kwargs: Any,
+    ) -> Marqo:
+        """Return Marqo initialized from texts. Note that Marqo does not need embeddings, we retain the
+        parameter to adhere to the Liskov substitution principle.
+
+        This is a quick way to get started with marqo - simply provide your texts and metadatas and this
+        will create an instance of the data store and index the provided data.
+
+        To know the ids of your documents with this approach you will need to include them in under the key "_id"
+        in your metadatas for each text
+
+        Example:
+        .. code-block:: python
+
+                from langchain.vectorstores import Marqo
+
+                datastore = Marqo(texts=['text'], index_name='my-first-index', url='http://localhost:8882')
+
+        Args:
+            texts (List[str]): A list of texts to index into marqo upon creation.
+            embedding (Any, optional): Embeddings (not required). Defaults to None.
+            index_name (str, optional): The name of the index to use, if none is provided then one will be created with a UUID. Defaults to None.
+            url (str, optional): The URL for Marqo. Defaults to "http://localhost:8882".
+            api_key (str, optional): The API key for Marqo. Defaults to "".
+            metadatas (Optional[List[dict]], optional): A list of metadatas, to accompany the texts. Defaults to None.
+            marqo_device (str, optional): The device for the marqo to use on the server, this is only used when a new index is being created. Defaults to "cpu".
```

Comment:
if it's not too many could be worth listing the options

Reply:
Added the cpu/cuda options to docstring